File: Data Structures/stack.js
class Node {
	constructor(value){
		this.value = value
		this.next = null
	}
}

class Stack {
	constructor(){
		this.top = null
		this.bottom = null
		this.length = 0
	}
	peek(){
		return this.top
	}
	isEmpty(){
		return Boolean(!this.length)
	}
	push(value){
		let newNode = new Node(value)
		if (this.length === 0) {
			this.top = newNode
			this.bottom = newNode
		} else {
			newNode.next = this.top
			this.top = newNode
		}
		this.length++
		return this
	}
	pop(){
		if (this.length === 0) {
			return "Nothing to pop"
		}

		let poppped = this.top
		if (this.length === 1) {
			this.top = null
			this.bottom = null
		} else {
			this.top = this.top.next
		}
		this.length--
		return poppped.value
	}
}

let myStack = new Stack()

File: Data Structures/stack.py
class Node:
	def __init__(self,value):
		self.value = value
		self.next = None

class Stack:
	def __init__(self):
		self.top = None
		self.bottom = None
		self.length = 0

	def isEmpty(self):
		return bool(self.length is 0)

	def peek(self):
		return self.top

	def push(self, value):
		newNode = Node(value)
		if (self.length == 0):
			self.top = newNode
			self.bottom = newNode
		else:
			newNode.next = self.top
			self.top = newNode
		self.length += 1
		return self

	def pop(self):
		if (self.length == 0):
			return "Nothing to return"

		popped = self.top
		if (self.length == 1):
			self.top = None
			self.bottom = None
		else:
			self.top = self.top.next

		self.length -= 1
		return popped


myStack = Stack()
